src/tests/payment/test_stripe_webhook.py

```diff
@@ -0,0 +1,430 @@
+def checkout_session_completed_payload():
+    """Fixture to generate a 'checkout.session.completed' event payload."""
+    return {
+        "type": "checkout.session.completed",
+        "data": {
+            "object": {
+                "id": "cs_test_a1SdT2v81sae3cW",
+                "object": "checkout.session",
+                "amount_subtotal": 990,
+                "amount_total": 990,
+                "billing_address_collection": "required",
+                "cancel_url": "http://127.0.0.1:8000/api/v1/subscriptions/cancel/",
+                "created": 1731848739,
+                "currency": "eur",
+                "custom_fields": [],
+                "customer": "cus_REbNQXKKFCRF2c",
+                "customer_creation": "always",
+                "customer_details": {
+                    "address": {
+                        "city": "Saint-Ouen-l'Aumône",
+                        "country": "FR",
+                        "line1": "1 Rue Pagnere",
+                        "line2": "null",
+                        "postal_code": "95310",
+                        "state": "null",
+                    },
+                    "email": "kadi@test.com",
+                    "name": "kadi",
+                },
+                "customer_email": "kadi@test.com",
+                "expires_at": 1731935139,
+                "metadata": {"product_name": "Essentiel", "user_id": "5"},
+                "mode": "subscription",
+                "payment_status": "paid",
+                "status": "complete",
+            },
+        },
+    }
+
+
+def generate_stripe_signature(checkout_session_completed_payload, secret):
+    """
+    Simulates the Stripe signature for the test using the secret key.
+    """
+    payload_str = json.dumps(checkout_session_completed_payload)
+    timestamp = str(int(time.time()))
+    sig_base = f"{timestamp}.{payload_str}"
+    signature = hmac.new(
+        secret.encode("utf-8"), sig_base.encode("utf-8"), hashlib.sha256
+    ).hexdigest()
+    return f"t={timestamp},v1={signature}"
+
+
+@pytest.mark.django_db
+class TestStripeWebhook:
+    url = reverse_lazy("v1:payment:stripe-webhook")
+
+    @mock.patch("stripe.Webhook.construct_event")
+    def test_checkout_session_completed_with_valid_signature(
+        self, mock_construct_event, client, checkout_session_completed_payload, user
+    ):
+        checkout_session_completed_payload["data"]["object"]["metadata"]["user_id"] = (
+            str(user.id)
+        )
+
+        sig_header = generate_stripe_signature(
+            checkout_session_completed_payload, STRIPE_WEBHOOK_SECRET
+        )
+        mock_construct_event.return_value = checkout_session_completed_payload
+        response = client.post(
+            self.url,
+            data=json.dumps(checkout_session_completed_payload),
+            content_type="application/json",
+            HTTP_STRIPE_SIGNATURE=sig_header,
+        )
+        assert response.status_code == status.HTTP_200_OK
```

Comment:
Perfectly valid approach of mock-patching and in fact it's what people use the most.
But I like to go one step further and patch only the line that needs to be patched using a context manager, e.g.
```
    def test_checkout_session_completed_with_valid_signature(
        self, client, checkout_session_completed_payload, user
    ):
        checkout_session_completed_payload["data"]["object"]["metadata"]["user_id"] = (
            str(user.id)
        )

        sig_header = generate_stripe_signature(
            checkout_session_completed_payload, STRIPE_WEBHOOK_SECRET
        )
        mock_construct_event = mock.Mock(return_value=checkout_session_completed_payload)
        with mock.patch("stripe.Webhook.construct_event", mock_construct_event):
            response = client.post(
                self.url,
                data=json.dumps(checkout_session_completed_payload),
                content_type="application/json",
                HTTP_STRIPE_SIGNATURE=sig_header,
            )
        assert response.status_code == status.HTTP_200_OK
```
Gave an example above, try to apply here and on the other tests

Reply:
I modified the test to follow your suggestion by using a mock.patch context directly on the specific line to be patched.